### PR TITLE
Add unit tests for ToolStripContentPanelDesigner

### PR DESCRIPTION
**Pull request by Syareel-Sukeri**
Related https://github.com/dotnet/winforms/issues/10773

## Proposed changes

- Add unit test ToolStripContentPanelDesignerTests.cs for public properties and method of the ToolStripContentPanelDesigner.
- Enable nullability in ToolStripContentPanelDesignerTests.cs.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12366)

### CODE REVIEW

**Review by Epica3055 — COMMENTED** [review]
looks good 👍

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContentPanelDesignerTests.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public void CanBeParentedTo_ShouldReturnFalse_WhenParentDesignerIsNotNull()
+    {
+        PanelDesigner parentDesigner = new();
+
+        bool result = _toolStripContentPanelDesigner.CanBeParentedTo(parentDesigner);
+
+        result.Should().BeFalse();
+    }
+
+    public void Dispose()
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContentPanelDesignerTests.cs:60]: Could you please more Dispose method right after the constructor, we do it in other classes.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContentPanelDesignerTests.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        bool result = _toolStripContentPanelDesigner.CanBeParentedTo(parentDesigner: null!);
+
+        result.Should().BeFalse();
+    }
+
+    [Fact]
+    public void CanBeParentedTo_ShouldReturnFalse_WhenParentDesignerIsNotNull()
+    {
+        PanelDesigner parentDesigner = new();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContentPanelDesignerTests.cs:53]:
```suggestion
        using PanelDesigner parentDesigner = new();
```

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, added minor comments

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContentPanelDesignerTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        _toolStripContentPanelDesigner = new ToolStripContentPanelDesigner();
+        _toolStripContentPanel = new ToolStripContentPanel();
+        _toolStripContentPanelDesigner.Initialize(_toolStripContentPanel);
+    }
+
+    [Fact]
+    public void SnapLines_ShouldReturnNonNullList()
+    {
+        var snapLines = _toolStripContentPanelDesigner.SnapLines;
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContentPanelDesignerTests.cs:25]: Please declare variables explicitly. This happens in other parts of the the code as well.

**Review by ricardobossan — APPROVED** [review]

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**